src/compose/service.coffee

```diff
@@ -661,6 +667,10 @@ module.exports = class Service
 		_.isEmpty(_.xor(_.keys(@networks), _.keys(otherService.networks)))
 
 	isSameContainer: (otherService) =>
+		# We need computed fields to be present to compare two services
+		@portMappings ?= @getPortsAndPortBindings()
```

Comment:
These two lines shouldn't be necessary if they're generated in the service constructor... and I'd very much prefer it if this function didn't mutate the services (keep it pure)